lib/Runtime/Library/InJavascript/Math_object.js

```diff
@@ -49,14 +51,18 @@
 
         for (let i = 1; i < arguments.length; i++) {
             nextVal = +arguments[i];
-            if (nextVal !== nextVal) {
-                return NaN;
+            if (nextVal !== nextVal || returnNaN) {
+                returnNaN = true;
```

Comment:
To very slightly optimise, please can you add `min = NaN` inside this `if` and then you can delete the `if` on line 62 (as after `returnNaN` is set min will not be changed again)